module/zfs/spa_misc.c

```diff
@@ -2019,6 +2019,236 @@ spa_dirty_data(spa_t *spa)
 	return (spa->spa_dsl_pool->dp_dirty_total);
 }
 
+/*
+ * ==========================================================================
+ * SPA Import Progress Routines
+ * ==========================================================================
+ */
+
+typedef struct spa_import_progress {
+	uint64_t		pool_guid;	/* unique # for updates */
+	char			*pool_name;
+	spa_load_state_t	spa_load_state;
+	uint64_t		check_start;	/* UTC time MMP check start */
+	uint64_t		check_end;	/* UTC time MMP check end */
+	uint64_t		spa_load_max_txg;	/* rewind txg */
+	procfs_list_node_t	smh_node;
+} spa_import_progress_t;
+
+spa_history_list_t *spa_import_progress_list = NULL;
+unsigned int spa_import_progress_list_size = 16;
```

Comment:
Why is it important to limit the number of entries to 16?  I understand this has no effect today since concurrent pool imports are not allowed, but I wonder if we want a limit at all.

Reply:
Since we remove entries from the list whenever an import ends, the list will not stay large, even if it grows large, so we do not need a limit.  And the semantic that there is one and only one entry in the procfile for each in-progress import is easy to describe.  

So I think you are right we do not want a limit.  I removed it.